TrackletAlgorithm/MatchProcessor.h

```diff
@@ -786,20 +1450,48 @@ void MatchProcessor(BXType bx,
       // bits of zproj, adjust the value up and down by zbins_adjust (2), then truncate the
       // zbins_adjust (2) LSBs to get the lower & upper bins that we need to look in.
       
+
       ap_int<6> zbin6 = izproj.range(izproj.length()-1,izproj.length()-MEBinsBits-zbins_nbitsextra);
+      //The -1 here is due to not using the full range of bits. Should be fixed.
+      ap_uint<8> rbin8 = izproj >> (izproj.length()-8-1);//izproj.range(izproj.length()-1,izproj.length()-8-1);
 
       //The first and last zbin the projection points to
-      ap_uint<MEBinsBits> zfirst, zlast;
-
+      constexpr int nZbinBits = ProjectionRouterBuffer<VMPTYPE, APTYPE>::kPRBufferZBinSize - 1; // zbin = (zfirst,zflag)
+      ap_uint<nZbinBits> zfirst, zlast, rtmp, rfirst;
+      ap_uint<1> rsecond;
+
+      ap_uint<nRbinBits> rbin = rbinLUT[rbin8];
+      typename VMProjection<VMPTYPE>::VMPFINEZ finer = rbin >> nZbinBits;
+      rtmp = rbin.range(nZbinBits, 0);
+      rfirst = rtmp >> 1;
+      rsecond = rtmp & 1;
+      rfirst = (izder.range(izder.length()-1, izder.length()-1)  == 1) ? ap_uint<nZbinBits>(rfirst + (1<<MEBinsBits)) : rfirst;
       (zfirst, zlast) = zbinLUT[(psseed,zbin6)];
 
-      typename VMProjection<VMPTYPE>::VMPZBIN zbin = (zfirst, zfirst!=zlast);      
+      typename VMProjection<VMPTYPE>::VMPZBIN zbin = (LAYER < trklet::N_LAYER) ? (zfirst, zfirst!=zlast) : (rfirst,rsecond);
 
       // VM Projection
       typename VMProjection<VMPTYPE>::VMPFINEZ finez = ((1<<(MEBinsBits+2))+(izproj>>(izproj.length()-(MEBinsBits+3))))-(zfirst,ap_uint<3>(0));
+      finez = (LAYER < trklet::N_LAYER) ?  finez : finer;
       
       //Extracts the rinv of the projection from the phider; recall phider = - rinv/2
       typename VMProjection<VMPTYPE>::VMPRINV rinv = (1<<(nbits_maxvm-1)) - 1 - iphider.range(iphider.length()-1,iphider.length()-nbits_maxvm);
+      if (LAYER >= trklet::N_LAYER) {
```

Comment:
Indentations?

Reply:
I'm not sure I see the issue

Reply:
How the code was indented looked odd to me earlier, but now when I look again it is fine...